src/libopenrave/iksolver.cpp

```diff
@@ -84,18 +84,25 @@ void IkFailureInfo::SaveToJson(rapidjson::Value& rIkFailureInfo, rapidjson::Docu
     if( !_description.empty() ) {
         orjson::SetJsonValueByKey(rIkFailureInfo, "description", _description, alloc);
     }
-    if( !_rCustomData.IsNull() && _rCustomData.MemberCount() > 0 ) {
-        rapidjson::Value rCustomDataCopy;
-        rCustomDataCopy.CopyFrom(_rCustomData, alloc);
-        rIkFailureInfo.AddMember("mapdata", rCustomDataCopy, alloc);
+    if (_mapCustomData.GetSize() > 0) {
+        rapidjson::Value rCustomData(rapidjson::kArrayType);
```

Comment:
Why not save `rCustomData` as an object? Should be easier for python to access, and we don't have to worry about speed/memory too much.

Reply:
I think json only accepts key as string, and here `nameId` is integer, so cannot make this `rCustomData` an object.